p2p/src/main/kotlin/maru/p2p/P2PNetworkFactory.kt

```diff
@@ -0,0 +1,176 @@
+/*
+   Copyright 2025 Consensys Software Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package maru.p2p
+
+import io.libp2p.core.ConnectionHandler
+import io.libp2p.core.Host
+import io.libp2p.core.PeerId
+import io.libp2p.core.crypto.PrivKey
+import io.libp2p.core.dsl.host
+import io.libp2p.core.multiformats.Multiaddr
+import io.libp2p.core.mux.StreamMuxerProtocol
+import io.libp2p.etc.types.millis
+import io.libp2p.pubsub.PubsubApiImpl
+import io.libp2p.pubsub.gossip.Gossip
+import io.libp2p.pubsub.gossip.GossipPeerScoreParams
+import io.libp2p.pubsub.gossip.GossipScoreParams
+import io.libp2p.pubsub.gossip.GossipTopicsScoreParams
+import io.libp2p.pubsub.gossip.builders.GossipParamsBuilder
+import io.libp2p.pubsub.gossip.builders.GossipRouterBuilder
+import io.libp2p.security.secio.SecIoSecureChannel
+import io.libp2p.transport.tcp.TcpTransport
+import java.util.Optional
+import kotlin.random.Random
+import org.apache.tuweni.bytes.Bytes
+import org.hyperledger.besu.metrics.noop.NoOpMetricsSystem
+import pubsub.pb.Rpc
+import tech.pegasys.teku.infrastructure.async.AsyncRunnerFactory
+import tech.pegasys.teku.infrastructure.async.MetricTrackingExecutorFactory
+import tech.pegasys.teku.infrastructure.unsigned.UInt64
+import tech.pegasys.teku.networking.p2p.libp2p.LibP2PNetwork
+import tech.pegasys.teku.networking.p2p.libp2p.LibP2PNodeId
+import tech.pegasys.teku.networking.p2p.libp2p.PeerManager
+import tech.pegasys.teku.networking.p2p.libp2p.gossip.GossipTopicHandlers
+import tech.pegasys.teku.networking.p2p.libp2p.gossip.LibP2PGossipNetwork
+import tech.pegasys.teku.networking.p2p.libp2p.gossip.PreparedPubsubMessage
+import tech.pegasys.teku.networking.p2p.libp2p.rpc.RpcHandler
+import tech.pegasys.teku.networking.p2p.network.P2PNetwork
+import tech.pegasys.teku.networking.p2p.network.PeerHandler
+import tech.pegasys.teku.networking.p2p.peer.Peer
+import tech.pegasys.teku.networking.p2p.reputation.ReputationManager
+
+object P2PNetworkFactory {
+  fun build(
+    privateKey: PrivKey,
+    port: String,
+    ipAddress: String,
+  ): P2PNetwork<Peer> {
+    val addr = Multiaddr("/ip4/$ipAddress/tcp/$port")
+
+    return setupP2PNetwork(privateKey, addr)
+  }
+
+  private fun setupP2PNetwork(
+    privateKey: PrivKey,
+    ipv4Address: Multiaddr,
+  ): P2PNetwork<Peer> {
+    val rpcMethod = MaruRpcMethod()
```

Comment:
I think this has to be moved away from here and `RpcHandler` probably has to be injected here

Reply:
RpcHandler is a Teku class, it uses RpcMethods to encode and decode messages. I guess once we know how we want to do it, we can have one or many RpcMethods to handle our messages.
I would want to make these changes once this PR is merged.